### PR TITLE
Enable auditing for create and amend case

### PR DESCRIPTION
**Pull request by chandaksweta**

### CODE REVIEW

File: app/uk/gov/hmrc/nationaldutyrepaymentcenter/controllers/ClaimController.scala [hunk 1]
```diff
@@ -66,54 +67,82 @@ class ClaimController @Inject()(
...
-                Json.toJson(
-                  NDRCCreateCaseResponse(
-                    correlationId = correlationId,
-                    error = Some(
-                      ApiError(
-                        errorCode = error.errorCode.getOrElse("ERROR_UPSTREAM_UNDEFINED"),
-                        errorMessage = error.errorMessage
-                      )
-                    )
+            if (error.isDuplicateCaseError) {
```
Comment (lmugnai) [inline app/uk/gov/hmrc/nationaldutyrepaymentcenter/controllers/ClaimController.scala]: @chandaksweta we don't have duplicate case errors on NDRC. Can we remove this?

Comment (chandaksweta) [reply app/uk/gov/hmrc/nationaldutyrepaymentcenter/controllers/ClaimController.scala]: For createCase what happens if the user goes back in the browser and submits the same request again??

Comment (chandaksweta) [reply app/uk/gov/hmrc/nationaldutyrepaymentcenter/controllers/ClaimController.scala]: and again...

File: app/uk/gov/hmrc/nationaldutyrepaymentcenter/services/AuditService.scala [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+                           ): JsObject =
+    Json.obj(
+      "success" -> caseResponse.isSuccess
+    ) ++
+      (if (caseResponse.isSuccess)
+        Json.obj(
+          "caseReferenceNumber" -> caseResponse.result.get.caseId
+        )
+      else
+        (if (reportDuplicate)
```
Comment (lmugnai) [inline app/uk/gov/hmrc/nationaldutyrepaymentcenter/services/AuditService.scala]: Should we remove this too?

File: app/uk/gov/hmrc/nationaldutyrepaymentcenter/models/responses/NDRCCaseResponse.scala [hunk 3]
```diff
@@ -27,7 +27,7 @@ case class NDRCCreateCaseResponse(
   def isDuplicate: Boolean = error.exists(_.errorCode == "409")
```
Comment (lmugnai) [inline app/uk/gov/hmrc/nationaldutyrepaymentcenter/models/responses/NDRCCaseResponse.scala]: Can we remove this?

File: app/uk/gov/hmrc/nationaldutyrepaymentcenter/services/AuditService.scala [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        (requestDetails ++ pegaResponseToDetails(updateResponse, false))
+    }
+
+    implicit val formats: Format[UpdateCaseAuditEventDetails] =
+      Json.format[UpdateCaseAuditEventDetails]
+  }
+
+  def pegaResponseToDetails(
+                             caseResponse: NDRCCaseResponse,
+                             reportDuplicate: Boolean
```
Comment (lmugnai) [inline app/uk/gov/hmrc/nationaldutyrepaymentcenter/services/AuditService.scala]: Can we remove?

**Review by lmugnai — APPROVED** [review]

**Pull request merged**